python/tank_vendor/shotgun_authentication/login_dialog.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Login dialog for authenticating to a Shotgun server.
+"""
+
+from .ui import resources_rc
+from .ui import login_dialog
+from . import session_cache
+from .errors import AuthenticationError
+from .ui.qt_abstraction import QtGui, QtCore
+
+
+class LoginDialog(QtGui.QDialog):
+    """
+    Dialog for getting user crendentials.
+    """
+    def __init__(self, title, is_session_renewal, hostname="", login="", fixed_host=False, http_proxy=None, parent=None):
+        """
+        Constructs a dialog.
+
+        :param title: Title of this dialog.
+        :param is_session_renewal: Boolean indicating if we are renewing a session or authenticating a user from scratch.
+        :param hostname: The string to populate the site field with. Defaults to "".
+        :param login: The string to populate the login field with. Defaults to "".
+        :param fixed_host: Indicates if the hostname can be changed. Defaults to False.
+        :param http_proxy: The proxy server to use when testing authentication. Defaults to None.
+        :param parent: The Qt parent for the dialog (defaults to None)
+        """
+        QtGui.QDialog.__init__(self, parent)
+
+        # setup the gui
+        self.ui = login_dialog.Ui_LoginDialog()
+        self.ui.setupUi(self)
+
+        # Set the title
+        self.setWindowTitle(title)
+
+        # Assign credentials
+        self._http_proxy = http_proxy
+        self.ui.site.setText(hostname)
+        self.ui.login.setText(login)
+
+        # default focus
+        if self.ui.site.text():
+            if self.ui.login.text():
+                self.ui.password.setFocus()
+            else:
+                self.ui.login.setFocus()
+        else:
+            self.ui.site.setFocus()
+
+        # set the logo
+        self.ui.logo.setPixmap(QtGui.QPixmap(":/shotgun_authentication/shotgun_logo_light_medium.png"))
+
+        # Disable keyboard input in the site and login boxes if we are simply renewing the session.
+        self.ui.site.setReadOnly(is_session_renewal or fixed_host)
+        self.ui.login.setReadOnly(is_session_renewal)
+
+        if is_session_renewal:
+            self._set_error_message("Your session has expired. Please enter your Shotgun password.")
```

Comment:
With the big Shotgun logo already present, maybe it's enough to say 

"Your session has expired. Please enter your password." ?